include/aws/http/http2_stream_manager.h

```diff
@@ -105,6 +105,26 @@ struct aws_http2_stream_manager_options {
     void *shutdown_complete_user_data;
     aws_http2_stream_manager_shutdown_complete_fn *shutdown_complete_callback;
 
+    /**
+     * Optional.
+     * When set, connection will be closed if 5xx response received from server.
+     */
+    bool close_connection_on_server_error;
+    /**
+     * Optional.
+     * The period for all the connections held by stream manager to send a PING in seconds.
+     * If you specify 0, manager will NOT send any PING.
+     * Note: if set, it must be large than the time of ping timeout setting.
+     */
+    size_t connection_ping_period_sec;
```

Comment:
I can't decide if this logic should live in the connection or not :|
It would be a bit cleaner there, since the h2 stream manager could give a connection back to the connection-mgr, then acquire it again and kick off more pings.

but then I don't want the h2_connection.c code any more complex than it already is so ... 🤷‍♀️

Reply:
I think it's okay to stay this level. I can see it may lead to a confusion that the ping for same connection was sent with the same time interval in the future. 

But, the idea for this is as the stream manager user, you don't have access to the connection under the hood. But if people can access connection, they can implement their own logic.

And we also said `The period for all the connections held by stream manager` in the documentation. So, user knows the time interval will only work if the connection is held by the stream manager all the time.